lib/chkname.c

```diff
@@ -58,6 +63,21 @@ login_name_max_size(void)
 static bool
 is_valid_name(const char *name)
 {
+	if (streq(name, "")
+	 || streq(name, ".")
+	 || streq(name, "..")
+	 || strcaseeq(name, "none")
+	 || strcaseeq(name, "all")
+	 || strcaseeq(name, "except")
+	 || strprefix(name, "-")
```

Comment:
The `+` character at beginning of name is also special: https://sourceware.org/git/?p=glibc.git;a=blob;f=nss/nss_compat/compat-pwd.c;h=7da6dc99f7e8aac6d212df42cc33221bdc6e2f35;hb=HEAD#l579

Reply:
The `!` should also be discouraged: https://github.com/linux-pam/linux-pam/blob/master/modules/pam_time/time.conf.5.xml#L64-L68

And I think that's all I can remember that I have seen as special cases. In fact, the `&` one was new to me. Discovered while searching for the `;` separators.